kubetest/main.go

```diff
@@ -512,6 +509,11 @@ func prepareGcp(o *options) error {
 	if err := migrateGcpEnvAndOptions(o); err != nil {
 		return err
 	}
+	if o.provider == "gke" && o.gcpNodeImage != "" {
+		// GKE needs both KUBE_NODE_OS_DISTRIBUTION and KUBE_GKE_IMAGE_TYPE
```

Comment:
Looking at this again, it looks like both GCE and GKE need `NODE_OS_DISTRIBUTION` set. `KUBE_NODE_OS_DISTRUBTION` will get into `NODE_OS_DISTRIBUTION` if it's run through bash, but the new GKE deployment doesn't handle it, for instance.